src/services/typeorm.ts

```diff
@@ -50,8 +56,59 @@ export class TypeormWrapper {
     return await this.connection.query(query);
   }
 
+  async maybeAttachPrimaryColumn(entity: EntityTarget<any>, value: any) {
+    // Grab the TypeORM metadata to determine the primary column of this entity and any join tables
+    const ormMetadata = getMetadataArgsStorage();
+    // Depth-first recursively add the primary column to nested entities, if they exist
+    for (const relation of ormMetadata.relations) {
+      if (relation.target !== entity) continue; // relations not involving this entity are ignored
+      if ((typeof relation.type) === 'string') continue; // "basic" SQL types won't have an id
+      if ((typeof relation.type) === 'function') {
+        const otherEntity = (relation.type as Function)(); // Why did TS fail the typecheck here?
+        const otherVal = value[relation.propertyName];
+        if (!!otherEntity && !!otherVal) {
+          await this.maybeAttachPrimaryColumn(otherEntity, otherVal);
+        }
+      }
+    }
+    // Now find the primary column of this entity (if one exists)
+    const primaryColumn = ormMetadata
+      .columns
+      .filter(c => c.target === entity)
+      .filter(c => c.options.primary)
+      .map(c => c.propertyName)
+      .shift();
```

Comment:
This manipulation of TypeORM's metadata may *seem* untyped, but TypeORM has fully typed these objects so the compiler will complain if you do the wrong thing with them, so changes to this metadata in TypeORM in the future should cause this code to fail to compile, and we can revisit how to adjust it without accidentally deploying broken code to production.